coap-service/coap_service_api.h

```diff
@@ -106,38 +86,58 @@ typedef int coap_service_security_start_cb(int8_t service_id, uint8_t EUI64[stat
  * CoAP service security done callback function.
  *
  * \param service_id         Id number of the current service.
- * \param EUI64              64 bit global identifier
+ * \param address            Address of sender
  * \param keyblock           Security key (40 bits)
  *
- * \return 0 for success / -1 for failure 
+ * \return 0 for success / -1 for failure
  */
-typedef int coap_service_security_done_cb(int8_t service_id, uint8_t EUI64[static 8], uint8_t keyblock[static 40]);
-
+typedef int coap_service_security_done_cb(int8_t service_id, uint8_t address[static 16], uint8_t keyblock[static 40]);
 
 /**
- * \brief Security service start callback
+ * \brief Initialise server instance.
  *
- * Starts security service handling.
+ * Initialise Thread services for the registered application.
  *
- * \param EUI64              64 bit global identifier
- * \param PSKd_ptr           Pointer to PSK key.
- * \param PSKd_len           Lenght of PSK key.
+ * \param interface_id       Informs registered application interface id. This parameter is passed to socket implementation.
+ * \param listen_port        Port that Application wants to use for communicate with coap server.
+ * \param service_options    Options of the current service.
+ * \param *start_ptr         Callback to inform security handling is started and to fetch device password.
+ * \param *security_done_cb  Callback to inform security handling is done.
  *
- * \return 0 for success / -1 for failure 
+ *  \return service_id / -1 for failure
  */
-int coap_service_security_key_set(int8_t service_id, uint8_t EUI64[static 8], uint8_t *PSKd_ptr, uint8_t PSKd_len);
+extern int8_t coap_service_initialize(int8_t interface_id, uint16_t listen_port, uint8_t service_options, coap_service_security_start_cb *start_ptr, coap_service_security_done_cb *security_done_cb);
 
 /**
- * \brief Service message response receive callback.
+ * \brief Service delete
  *
- * Function that handles CoAP service message receiving and parsing
+ * Removes all data related to this instance
  *
- * \param msg_id           Id number of the current message.
- * \param response_ptr     Pointer to CoAP header structure.
+ * \param service_id         Id number of the current service.
+ */
+extern void coap_service_delete( int8_t service_id );
+
+/**
+ * \brief Close secure connection
  *
- * \return 0 for success / -1 for failure 
-  */
-typedef int coap_service_response_recv(int8_t service_id, uint16_t msg_id, sn_coap_hdr_s *response_ptr);
+ * Closes secure connection (if present), but leaves socket open.
+ *
+ * \param service_id         Id number of the current service.
+ */
+extern void coap_service_close_secure_connection(int8_t service_id);
```

Comment:
this needs remote address and port to close one specific session one service can have many sessions and might want to close only one